### PR TITLE
Add a test case for playing a short sound from an entity using virtual baton

### PR DESCRIPTION
**Pull request by ericrius1**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 918c61c166f622277e8b13b87b963928a2982767 available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

File: examples/libraries/virtualBaton.js [hunk 1]
```diff
@@ -371,7 +371,7 @@ function virtualBatonf(options) {
     };
 
     // Gather nAcceptors by making two proposals with some gathering time, even without a claim.
-    propose();
+   propose();
```
Comment (howard-stearns) [inline examples/libraries/virtualBaton.js]: Introduces bad whitespace.

File: examples/tests/basicEntityTest/entitySpawner.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+  var orientation = Camera.getOrientation();
+  orientation = Quat.safeEulerAngles(orientation);
+  orientation.x = 0;
+  orientation = Quat.fromVec3Degrees(orientation);
+  var center = Vec3.sum(MyAvatar.position, Vec3.multiply(3, Quat.getFront(orientation)));
+
+  // Math.random ensures no caching of script
+  var SCRIPT_URL = Script.resolvePath("myEntityScript.js?v1" + Math.random())
```
Comment (howard-stearns) [inline examples/tests/basicEntityTest/entitySpawner.js]: I understand doing this development, but should we be playing cache-defeating games in release/deployed/example code?

(And if we do decide we need to, shouldn't we at least do proper name=value query parameter?)

File: examples/tests/batonSoundEntityTest/batonSoundTestEntityScript.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
+
+(function() {
+    Script.include("../../libraries/virtualBaton.js?v2" + Math.random());
```
Comment (howard-stearns) [inline examples/tests/batonSoundEntityTest/batonSoundTestEntityScript.js]: same cache-defeat query parameter question as above

**Comment by hifi-gustavo** [comment]
Build console output for commit: a0e8e00357f9aaf4e8c1aaba030349b1fb95b16f available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dfffe5f431bc682dc1d2b2d03bae8cb6e348a248 available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by ericrius1** [comment]
Going to rewrite based on discussion with philip- do not merge or test for now

**Comment by hifi-gustavo** [comment]
Build console output for commit: ad5a111b85c3517a60a2a30bea229fd70bcc0781 available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 608821b50b0896c682202f1652d83bf92a5bd45f available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 107479039ae18420b60d67c24b8128bd9d5fc091 available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 60bb77fd346afa0c42a2af86afc3fe473599ecb5 available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e5364fd186568c4c6606277e1b5f54d80e0572f0 available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8ce23e23451d1246910308d69c54f401bfe45fb4 available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7450a2ea3a1df369d291970165aeb2ea8e38a84b available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Comment by ericrius1** [comment]
To Test:

1) Gather two people together
2) Run https://rawgit.com/ericrius1/hifi/batonSoundEntityTest/examples/entityScripts/virtualBaton/batonSimpleEntitySpawner.js

2) You will see a box appear over the head of one of the two users, and a box appear a few meters in front of the client who ran the script. The box over the head represents the client who currently owns the baton entity script

3) The current owner should now switch domains or quit, and npw the box should now appear above the remaining user's head. In addition, the other box will change color and light up for a brief moment, signalling that ownership has been successfully transferred

**Comment by cozza13** [comment]
Tested with 2 and 3 people and it worked. QA PASS

**Comment by hifi-gustavo** [comment]
Build console output for commit: e261519b11930fe1c1895441d55e1de70ac2c2c4 available at: https://gist.github.com/55016742864acdfbbc71<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7424.dmg

**Pull request merged**